admin/services/data-access/pupil.data.service.js

```diff
@@ -46,6 +46,27 @@ pupilDataService.sqlFindPupilsWithStatusByDfeNumber = async function (dfeNumber)
   return sqlService.query(sql, [paramDfeNumber])
 }
 
+/**
+ * Fetch all pupils for a school by dfeNumber with their status codes and attendance reasons
+ * @param dfeNumber
+ * @returns {Promise<*>}
+ */
+pupilDataService.sqlFindPupilsWithStatusAndAttendanceReasons = async function (dfeNumber) {
+  const paramDfeNumber = { name: 'dfeNumber', type: TYPES.Int, value: dfeNumber }
+
+  const sql = `
+      SELECT p.*, ps.code, pg.group_id, ac.reason, ac.code as reasonCode
+      FROM ${sqlService.adminSchema}.${table} p
+      INNER JOIN school s ON s.id = p.school_id
+      LEFT JOIN pupilStatus ps ON (p.pupilStatus_id = ps.id)
+      LEFT OUTER JOIN ${sqlService.adminSchema}.[pupilAttendance] pa ON p.id = pa.pupil_id AND (pa.isDeleted IS NULL OR pa.isDeleted = 0)
```

Comment:
I would prefer the whole ON clause to be bracketed for readability

Reply:
Usually I even place the ON clause on the following line (indented) for even more readability.